### PR TITLE
Solving issue #23: randomising order of participants

### PR DESCRIPTION
**Pull request by falafel72**
Added client-side javascript to allow for randomising order of leaderboard.

### CODE REVIEW

File: static/js/scripts.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+window.onload = function() {
```
Comment (blazeu) [inline static/js/scripts.js]: No need to put on `onload`.
The script is already on the bottom of the page.

File: static/js/scripts.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+    // select all leaderboards
+    var allLeaders = document.querySelectorAll("div.orgLists ul");
+    allLeaders.forEach(function(orgList) {
+        //shuffle each leaderboard
```
Comment (blazeu) [inline static/js/scripts.js]: space after //

**Review by blazeu — CHANGES_REQUESTED** [review]
After you're done, run `coala` to format the code https://coala.io.

---

```
$(npm bin)/prettier --single-quote --no-semi --trailing-comma es5 --write static/js/scripts.js
```
^ if you can't run coala.

**Comment by blazeu** [comment]
Also, keep in mind to not push changes from your master branch - create a new branch instead. Might lead to some conflicts. Refer http://api.coala.io/en/latest/Developers/Newcomers_Guide.html#step-3-creating-a-fork-and-testing-your-changes. :smile:

**Comment by falafel72** [comment]
I'll be sure to keep that in mind 👍 
Everything should be fixed now.

**Comment by blazeu** [comment]
Nice. Please squash into a single commit 
https://api.coala.io/en/latest/Developers/Git_Basics.html#squashing-your-commits

**Review by blazeu — CHANGES_REQUESTED** [review]
Format your commit message according to https://coala.io/commit

**Comment by falafel72** [comment]
Things got a little awkward cause I originally cloned it and forgot to fork...should be fine now though.

**Comment by blazeu** [comment]
No, still 5 commit :/
You can ask in our chatroom @ https://coala.zulipchat.com/ if you're confused :)

**Comment by blazeu** [comment]
Added -> Add

**Comment by falafel72** [comment]
Just tried using `git reset --soft`, did that work?

**Comment by blazeu** [comment]
Run `git rebase -i HEAD~7`

7 is your commit count in this case.

Then, you want to replace `pick` to `squash`, except for the first one.

Hit save and tidy up your commit message so that it only contains your main commit.

**Comment by jayvdb** [comment]
Comment on 9a4648c91e8742e9e93ac348c758efa8f4ad4eea, file scrap.js, line 34.

Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.whitespace`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpo1u7h43c/scrap.js
+++ b/tmp/tmpo1u7h43c/scrap.js
@@ -31,7 +31,7 @@
 
 ;(async () => {
   const data = await fetchOrgsWithLeaders()
-  
+
   // sort data by completed_task_instance_count
   data.sort((a, b) => 
     b.completed_task_instance_count - a.completed_task_instance_count
```

**Comment by jayvdb** [comment]
Comment on 9a4648c91e8742e9e93ac348c758efa8f4ad4eea, file scrap.js, line 36.

Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.whitespace`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpo1u7h43c/scrap.js
+++ b/tmp/tmpo1u7h43c/scrap.js
@@ -33,7 +33,7 @@
   const data = await fetchOrgsWithLeaders()
   
   // sort data by completed_task_instance_count
-  data.sort((a, b) => 
+  data.sort((a, b) =>
     b.completed_task_instance_count - a.completed_task_instance_count
   )
 
```

File: package.json [hunk 3]
```diff
@@ -11,14 +11,16 @@
   "license": "MIT",
   "dependencies": {
     "mustache": "^2.3.0",
-    "ncp": "^2.0.0",
```
Comment (jayvdb) [inline package.json]: This file does not contain parsable JSON. Expecting ',' delimiter.

*Origin: JSONFormatBear, Section: `json`.*

**Comment by jayvdb** [comment]
Comment on 9a4648c91e8742e9e93ac348c758efa8f4ad4eea, file .travis.yml, line 10.

too few spaces before comment (comments)

*Origin: YAMLLintBear, Section: `yaml`.*

**Comment by jayvdb** [comment]
Comment on 9a4648c91e8742e9e93ac348c758efa8f4ad4eea, file .travis.yml, line 12.

truthy value is not quoted (truthy)

*Origin: YAMLLintBear, Section: `yaml`.*

**Comment by falafel72** [comment]
Seems like the problem is that I've got merge conflicts from merging my fork and my clone and I undo the merge every time I squash all my commits. Is there a way I can integrate the changes made by the merge into my squashed commits?

**Comment by blazeu** [comment]
Well I did tell the wrong instruction :disappointed: 

What you could do now is back up your modified files (just yours, not the others)

Add this repo as `upstream` remote and then `git fetch upstream`

Do `git reset --hard upstream/master` (this will remove your change, so make sure you have a backup).

Then apply your changes manually and commit.

Do not push until your new commit count is exactly 1 (check `git log`).

You should push with `-f` after this.

You can ask more here https://coala.zulipchat.com

File: static/js/app.js [hunk 4]
```diff
@@ -26,3 +26,12 @@ function timeDifference(current, previous) {
...
 }
+
+// select all leaderboards
+var allLeaders = document.querySelectorAll("div.org ul");
+  allLeaders.forEach(function(orgList) {
+  // shuffle each leaderboard
+  for(var i = orgList.children.length; i >= 0; i--) {
+    orgList.appendChild(orgList.children[Math.random()*i | 0]);
+  }
+});
```
Comment (jayvdb) [inline static/js/app.js]: Line contains following spacing inconsistencies:
- No newline at EOF.

*Origin: SpaceConsistencyBear, Section: `all.whitespace`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpd5h9drm4/static/js/app.js
+++ b/tmp/tmpd5h9drm4/static/js/app.js
@@ -34,4 +34,4 @@
   for(var i = orgList.children.length; i >= 0; i--) {
     orgList.appendChild(orgList.children[Math.random()*i | 0]);
   }
-});+});
```

**Comment by falafel72** [comment]
Thanks, looks like it's finally working.

**Comment by andrewda** [comment]
ack 3aafa26998d5505f12385ee8728b6eb913755ceb

**Review by blazeu — APPROVED** [review]

**Pull request merged**